### PR TITLE
Create single table Boundaries property

### LINKED ISSUE #391: Create single table Boundaries property
### Problem Description

Now that we have property classes, we want to move the metrics used in the `DiagnosticReport` into their respective properties.

This is the spec for the `Boundaries` property

#### Attributes

- `_details` - A dataframe containing the following columns
    - 'Column'
    - 'Metric'
    - 'Score'

The metrics used in this class are: [BoundaryAdherence]

#### Methods

- `_generate_details(self, real_data, synthetic_data, metadata, progress_bar)` - Generate the `_details` dataframe that will be used by the base class to get the score.
Pseudocode:
```python
for column in real_data:
    metric = 'BoundaryAdherence'
    score = BoundaryAdherence.computeBreakdown(real_data, synthetic_data)
```
- `get_visualization()` - Returns a plotly.graph_objects._figure.Figure object for the specified table. Use code similar to what's in [get_boundaries_plot](https://github.com/sdv-dev/SDMetrics/blob/master/sdmetrics/reports/single_table/plot_utils.py#L136).

### Expected behavior

```python
>>> prop = Synthesis()
>>> prop.get_score()
>>> 0.89
>>> prop.get_visualization()
```

### Additional context

- we should catch any exceptions the same way we do [here](https://github.com/sdv-dev/SDMetrics/blob/master/sdmetrics/reports/single_table/_properties/column_shapes.py)

### PR DESCRIPTION
**Pull request by R-Palazzo**
Resolve #391

### CODE REVIEW

File: tests/unit/reports/single_table/_properties/test_boundary.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import numpy as np
+import pandas as pd
+
+from sdmetrics.reports.single_table._properties.boundary import Boundary
+
+
+class TestBoundary:
+
+    @patch('sdmetrics.reports.single_table._properties.boundary.BoundaryAdherence.compute')
+    def test__generate_details(self, boundary_adherence_mock):
```
Comment (amontanez24) [inline tests/unit/reports/single_table/_properties/test_boundary.py:12]: can we add one unit test where the metric is mocked to throw an error so we can test the exception handling?

Comment (R-Palazzo) [reply tests/unit/reports/single_table/_properties/test_boundary.py:12]: Yes, done in [e260202](https://github.com/sdv-dev/SDMetrics/pull/400/commits/e260202e27f1610cbe8b540aa6b054d0e72f4851)

**Review by amontanez24 — APPROVED** [review]
One comment but otherwise looks good! Can you make a branch for `DiagnosticReport` issues and start merging these there?

**Review by pvk-developer — APPROVED** [review]
LGTM!

**Pull request merged**